src/main/java/com/backend/controller/PortfolioController.java

```diff
@@ -189,6 +191,33 @@ public GetAllAssetsByPortfolioIdResponse getAllAssetsByPortfolioId(@PathVariable
 		return response;
 	}
 
+	@GetMapping(path = "/portfolio/{pid}/transactions")
+	public List<Map<String, Object>> getTransactionsByPortfolioId(@PathVariable int pid) {
+		List<PortfolioAsset> portfolioAssetList = portfolioAssetService.findAllByPortfolioId(pid);
+		List<Map<String, Object>> transactionList = new ArrayList<>();
+
+		for (int i = 0; i < portfolioAssetList.size(); i++) {
+			long unixCreated = portfolioAssetList.get(i).getDateCreated();
+			long unixModified = portfolioAssetList.get(i).getDateModified();
+			Date created = new Date(unixCreated*1000L);
+			Date modified = new Date(unixModified*1000L);
+			SimpleDateFormat jdf = new SimpleDateFormat("yyyy-MM-dd HH:mm z");
+			jdf.setTimeZone(TimeZone.getTimeZone("GMT+8"));
+			String createdDate = jdf.format(created);
+			String modifiedDate = jdf.format(modified);
+			Map<String, Object> transaction = new HashMap<>();
+			transaction.put("portfolioAssetId", portfolioAssetList.get(i).getPortfolioAssetId());
+			transaction.put("portfolioId", portfolioAssetList.get(i).getPortfolioId());
+			transaction.put("assetId", portfolioAssetList.get(i).getAssetId());
+			transaction.put("averagePrice", portfolioAssetList.get(i).getAveragePrice());
+			transaction.put("quantity", portfolioAssetList.get(i).getQuantity());
+			transaction.put("dateCreated", createdDate);
+			transaction.put("dateModified", modifiedDate);
+			transactionList.add(transaction);
```

Comment:
nits: would it be better to declare a class for this? seems quite verbose

Reply:
Will leave it as it is for now since the data types returned are different from the original portfolioAsset class. A bit laze to created new classes LOL